src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -59,26 +59,30 @@ export class HydrateNewVehicleRecordComponent {
 
     this.technicalRecordService.editableVehicleTechRecord$
       .pipe(
-        withLatestFrom(this.technicalRecordService.batchVehicles$),
+        withLatestFrom(this.technicalRecordService.batchVehicles$, this.isBatch$),
         take(1),
-        map(([record, batch]) =>
-          (record ? [record] : []).concat(
-            batch.map(
-              v =>
-                ({
-                  ...record!,
-                  vin: v.vin,
-                  vrms: v.trailerId ? [{ vrm: v.trailerId, isPrimary: true }] : null,
-                  trailerId: v.trailerId ?? null
-                } as VehicleTechRecordModel)
-            )
-          )
-        )
+        map(([record, batch, isBatch]) => {
```

Comment:
May I offer an approach that does not necessitate modifying the map at all?

```typescript
    this.technicalRecordService.editableVehicleTechRecord$
      .pipe(
        withLatestFrom(this.technicalRecordService.batchVehicles$),
        take(1),
        map(([record, batch]) =>
          (record ? [record] : []).concat(
            batch.map(
              v =>
                ({
                  ...record!,
                  vin: v.vin,
                  vrms: v.trailerId ? [{ vrm: v.trailerId, isPrimary: true }] : null,
                  trailerId: v.trailerId ?? null
                } as VehicleTechRecordModel)
            )
          )
        ),
        withLatestFrom(this.technicalRecordService.batchVehicles$, this.isBatch$),
      )
      .subscribe(([vehicleList, isBatch]) => {
        vehicleList.forEach(vehicle => this.store.dispatch(createVehicleRecord({ vehicle })));

        this.navigateTo(isBatch ? 'batch-results' : '..');
      });
```

Reply:
I just added the `withLatestFrom(this.isBatch$)` after the `map()` and used the value in the subscription like you have.

Reply:
Yeah I didn't like the idea of two with latest from, but yes it's somewhat nicer 👍